testing/functional_tests_assertions.rst

```diff
@@ -0,0 +1,79 @@
+.. index::
+   single: Tests; Assertions
+
+Functional Test specific assertions Reference
+=============================================
+
+When doing functional tests, sometimes you need to make complex assertions in
+order to check whether the ``Request``, the ``Response`` or the ``Crawler``
+contain the expected information to make your test succeed.
+
+Here is an example without the specific assertions::
+
+    $this->assertGreaterThan(
+        0,
+        $crawler->filter('html:contains("Hello World")')->count()
+    );
+
+Now here is the example with the assertions introduced in Symfony 4.3::
+
+    $this->assertSelectorTextContains('html', 'Hello World');
+
+As you can see, even though the first assertion uses native PHPUnit methods,
+the second one explains more what we want to expect.
+
+As of Symfony 4.3, a list of assertions have been added, here is the reference.
+
+.. note::
+
+    These assertions will not work with `symfony/panther`_ as they use the
+    ``Request`` and ``Response`` objects coming from the ``HttpFoundation``
```

Comment:
I would remove the explanation about why they don't work with Panther (nobody cares). Wouldn't it be possible to make them work?

Reply:
Maybe not adding it directly here, but adding a Troubleshooting headline at the end and explain the panther case sounds helpful to me.

Reply:
To make them work we should make all assertions compatible with both `HttpFoundation` and `BrowserKit` classes (especially `Request` and `Response`), and make sure the `Browser` class used is also compatible with Panther.

Reply:
... which is quite possible, right? I suppose you meant "constraints" and not "assertions" here.

Reply:
Yep, I wasn't originally planning to use constraints, but since you updated the PR with it, we can make them compatible with both classes indeed 😄